benchmarks/benchmarks/loading.py

```diff
@@ -0,0 +1,186 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+File loading benchmark tests.
+
+Where applicable benchmarks should be parameterised for two sizes of input data:
+  * minimal: enables detection of regressions in parts of the run-time that do
+             NOT scale with data size.
+  * large: large enough to exclusively detect regressions in parts of the
+           run-time that scale with data size. Size should be _just_ large
+           enough - don't want to bloat benchmark runtime.
+
+"""
+
+from iris import AttributeConstraint, Constraint, load, load_cube
+from iris.cube import Cube
+from iris.fileformats.um import structured_um_loading
+
+from .generate_data import BENCHMARK_DATA, REUSE_DATA, run_function_elsewhere
+from .generate_data.um_files import create_um_files
+
+
+class LoadAndRealise:
+    params = [
+        [(2, 2, 2), (1280, 960, 5)],
```

Comment:
We've seen a few issues where a pp file with lots of layers is significantly less performant than one with a few - could we try another shape (if I understand what's here correctly) that's not very wide but quite deep?

Reply:
https://github.com/SciTools/iris/pull/4477#issuecomment-1025613155

Reply:
I'm happy that this shouldn't be done here, but think our benchmark coverage should (relatively soon) be in a place where it accounts for this.

Reply:
Yes, agreed